src/app/vm/vm-creation/template/agreement/vm-creation-agreement.component.ts

```diff
@@ -0,0 +1,72 @@
+import { Component, Inject, OnInit } from '@angular/core';
+import { MAT_DIALOG_DATA, MatDialog, MatDialogRef } from '@angular/material';
+import { Http } from '@angular/http';
+import { Observable } from 'rxjs/Observable';
+
+import { Converter } from 'showdown';
+import { TemplateTagService } from '../../../../shared/services/tags/template-tag.service';
+import { BaseTemplateModel } from '../../../../template/shared/base-template.model';
+import { UserTagService } from '../../../../shared/services/tags/user-tag.service';
+
+
+@Component({
+  selector: 'cs-vm-creation-template-agreement',
+  templateUrl: 'vm-creation-agreement.component.html'
+})
+export class VmCreationAgreementComponent implements OnInit {
+  private agreement: string;
+  private template: BaseTemplateModel;
+  private lang: string;
+
+  constructor(
+    @Inject(MAT_DIALOG_DATA) data,
+    private dialog: MatDialog,
+    private dialogRef: MatDialogRef<VmCreationAgreementComponent>,
+    private http: Http,
+    private templateTagService: TemplateTagService,
+    private userTagService: UserTagService
+  ) {
+    this.template = data;
+  }
+
+  public ngOnInit() {
+    this.userTagService.getLang().subscribe(res => this.lang = res);
+    this.readFile();
+  }
+
+  public onCancel() {
+    this.dialogRef.close();
+  }
+
+  public onAgree() {
+    this.dialogRef.close(true);
+  }
+
+  protected readFile() {
+    this.templateTagService.getAgreement(this.template).subscribe(path => {
+      let langPath;
```

Comment:
Let's modify this logic.
Right now it requests the file with translation, and if fails request it without. let's store different tags:
csui.template.agreement.ru for an Russian agreement
csui.template.agreement.en for an English agreement
csui.template.agreement for the default agreement, so, there will be only 1 call without guessing the path